src/Microsoft.DotNet.Build.Tasks/GenerateDepsJson.cs

```diff
@@ -110,52 +110,56 @@ public override bool Execute()
                 }
             }
 
-            targetsSection.Remove(runtimeTarget);
-            newDepsJson.Remove("libraries");
+            string platformPackageName = "";
+            foreach (var package in targetsSection[runtimeTarget].Children())
+            {
+                JProperty packageProperty = package as JProperty;
+                if (packageProperty != null && packageProperty.Name.Contains($"runtime.{rid}.Microsoft.NETCore.App"))
+                {
+                    platformPackageName = packageProperty.Name;
+                    break;
+                }
+            }
+
+            if (string.IsNullOrEmpty(platformPackageName))
+                Log.LogError($"Unable to find platform package that matches runtime.{rid}.Microsoft.NETCore.App when generating deps.json file.");
+
+            JObject platformPackage = (JObject)targetsSection[runtimeTarget][platformPackageName];
+            platformPackage.Remove("runtime");
+            platformPackage.Remove("native");
+            JObject platformRuntime = new JObject();
+            JObject platformNative = new JObject();
+
             foreach (var assembly in assemblyNames)
             {
-                JObject runtimes = new JObject();
-                JObject runtimeLocation = new JObject();
-                string key = $"{assembly.AssemblyName.Name}/{assembly.AssemblyName.Version.Major}.{assembly.AssemblyName.Version.Minor}.{assembly.AssemblyName.Version.Build}";
-                runtimeLocation.Add(assembly.FileName, new JObject());
-                runtimes.Add("runtime", runtimeLocation);
                 try
                 {
-                    targetValue.Add(key, runtimes);
+                    platformRuntime.Add($"runtimes/{rid}/lib/netcoreapp2.0/{assembly.FileName}", new JObject());
                 }
                 catch (System.ArgumentException)
                 {
 
                 }
+            }
+            // Add mscorlib and System facades to platformRuntime
+            platformRuntime.Add($"runtimes/{rid}/lib/netcoreapp2.0/mscorlib.dll", new JObject());
+            platformRuntime.Add($"runtimes/{rid}/lib/netcoreapp2.0/System.dll", new JObject());
 
+            foreach (var nativeAssembly in nonManagedAssemblyFilePaths)
+            {
                 try
                 {
-                    libraryValue.Add(key, ConstructLibraryNode());
+                    if (!nativeAssembly.EndsWith(".pdb"))
```

Comment:
Why do we need to check for pdb's here but not in the managed assembly case?

Reply:
nativeAssembly is basically a bag where we throw anything that is not a managed assembly, which means that all the pdbs are going here along with the native binaries. That is why we only have the check in the nativeAssemblies.